### PR TITLE
Convert af commands to rzshell

### PR DESCRIPTION
**Pull request by p371jah**
**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

- `af+` syntax was changed (uses `core->offset` as addr now)
- old handlers were removed, discovered incosistents in tests were fixed

Idk what to do with `afa` and `afj` (both untested)

...

**Test plan**

...

**Closing issues**

Related #1442

### CODE REVIEW

File: librz/core/cmd/cmd_analysis.c [hunk 1]
```diff
@@ -7061,6 +6811,127 @@ RZ_IPI RzCmdStatus rz_analysis_function_cc_reg_usage_handler(RzCore *core, int a
...
+		RZ_LOG_ERROR("Cannot add function (duplicated)\n");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_analysis_function_del_handler(RzCore *core, int argc, const char **argv) {
+	RzAnalysisFunction *f = analysis_get_function_in(core->analysis, core->offset);
+	if (!f) {
+		return RZ_CMD_STATUS_ERROR;
```
Comment (XVilka) [inline librz/core/cmd/cmd_analysis.c:6891]: Probably worth printing the error message as well.

**Review by XVilka — APPROVED** [review]

File: librz/core/cmd/cmd_analysis.c (hunk 1 above)
Comment (p371jah) [reply librz/core/cmd/cmd_analysis.c:6891]: `analysis_get_function_in` does it

**Comment by ret2libc** [comment]
For `afj` I found commit 3f6fb93689d6dbd337d64b6a72ffe68b6cf85267

File: librz/core/cmd/cmd_analysis.c [hunk 2]
```diff
@@ -7061,6 +6811,127 @@ RZ_IPI RzCmdStatus rz_analysis_function_cc_reg_usage_handler(RzCore *core, int a
...
+		}
+	}
+	rz_list_free(list);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_analysis_function_add_recu_handler(RzCore *core, int argc, const char **argv) {
+	const char *name = argc == 2 ? argv[1] : NULL;
+	bool analyze_recursively = true;
+	if (!strcmp(argv[0], "af")) {
```
Comment (ret2libc) [inline librz/core/cmd/cmd_analysis.c:6840]: can this be called from anywhere else?

File: librz/core/cmd/cmd_analysis.c [hunk 3]
```diff
@@ -7061,6 +6811,127 @@ RZ_IPI RzCmdStatus rz_analysis_function_cc_reg_usage_handler(RzCore *core, int a
...
+		analyze_recursively = rz_config_get_i(core->config, "analysis.calls");
+	}
+	bool result = rz_core_analysis_function_add(core, name, core->offset, analyze_recursively);
+	return result ? RZ_CMD_STATUS_OK : RZ_CMD_STATUS_ERROR;
```
Comment (ret2libc) [inline librz/core/cmd/cmd_analysis.c]:
```suggestion
	return bool2status(rz_core_analysis_function_add(core, name, core->offset, analyze_recursively));
```

**Review by ret2libc — COMMENTED** [review]
LGTM overall

File: librz/core/cmd/cmd_analysis.c (hunk 2 above)
Comment (p371jah) [reply librz/core/cmd/cmd_analysis.c:6840]: Yes, `afr`

Comment (pelijah) [reply librz/core/cmd/cmd_analysis.c:6840]: Btw I don't see a difference between `af` and `afr`. This `analysis.call` flag is really confusing. `af` internals anyway invoke `rz_analysis_analyze_fcn_refs`.

Comment (wargio) [reply librz/core/cmd/cmd_analysis.c:6840]: i suggest to check it via `git blame` often it helps

**Review by ret2libc — APPROVED** [review]

**Pull request merged**